presto-main/src/main/java/com/facebook/presto/operator/ExchangeClient.java

```diff
@@ -535,6 +542,17 @@ private static void closeQuietly(PageBufferClient client)
         }
     }
 
+    private static Optional<URI> getAsyncPageTransportLocation(URI location, boolean asyncPageTransportEnabled)
+    {
+        if (asyncPageTransportEnabled) {
+            String path = location.getPath().replace("v1/task", "v1/task/async");
```

Comment:
Instead of getting a `v1/task` and then rewriting it into `v1/task/async`, can we just return the `/v1/task/async` address in `HttpLocationFactory#createHttpTaskLocation`, cc @highker 

If there are places needs to use the old `v1/task` endpoint or this is not easy to done, just leave a comment why we need to rewrite the address :)

Reply:
`v1/task` endpoint is used for all other non-page-transport requests, I added comment to explain why two endpoints must co-exist.

Reply:
I had the same thought.  Ideally, once this is battle tested in production environments, we'll delete the synchronous endpoints, and can remove this then.